### PR TITLE
fix: process.env members could be undefined, abstract into `RustEnv` …

### PR DESCRIPTION
**Pull request by icyJoseph**
Hi,

Really cool project!

I've started to familiarize with the code base, so here are some type fixes for #47.

What do we think of upgrading to TS 4.4.x?

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -67,6 +70,10 @@ async function runUserScripts(entrypoint: string) {
   }
 }
 
+function isExecaError(err: Error): err is ExecaError {
```
Comment (icyJoseph) [inline src/index.ts:73]: Have to roll out our own type guard to make sure we are dealing with an `ExecaError`

File: src/index.ts [hunk 2]
```diff
@@ -79,10 +86,12 @@ async function cargoLocateProject(config: CargoConfig) {
...
   } catch (e) {
-    if (!/could not find/g.test(e.stderr)) {
-      console.error("Couldn't run `cargo locate-project`");
-      throw e;
+    if (e instanceof Error && isExecaError(e)) {
+      if (!/could not find/g.test(e.stderr)) {
+        console.error("Couldn't run `cargo locate-project`");
+      }
     }
+    throw e;
```
Comment (icyJoseph) [inline src/index.ts:94]: Should always re throw? I think?

File: src/index.ts (hunk 1 above)
Comment (leerob) [reply src/index.ts:73]: Thank you!

**Comment by leerob** [comment]
> What do we think of upgrading to TS 4.4.x?

That would be amazing 😄

**Review by leerob — APPROVED** [review]

**Pull request merged**